front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -11,16 +17,146 @@ import { useMemo } from "react";
+    SetStateAction<DataSourceViewSelectionConfigurations>
+  >;
+};
+
+export function DataSourceViewsSelector({
+  owner,
+  dataSourceViews,
+  selectionConfigurations,
+  setSelectionConfigurations,
+}: DataSourceViewsSelectorProps) {
+  const applyGrouping =
+    dataSourceViews.length >= MIN_TOTAL_DATA_SOURCES_TO_GROUP;
+
+  const groupManaged =
+    applyGrouping &&
+    dataSourceViews.filter((dsv) => isManaged(dsv.dataSource)).length >=
+      MIN_DATA_SOURCES_PER_KIND_TO_GROUP;
 
-export function DataSourceViewSelector({
+  const groupFolders =
+    applyGrouping &&
+    dataSourceViews.filter((dsv) => isFolder(dsv.dataSource)).length >=
+      MIN_DATA_SOURCES_PER_KIND_TO_GROUP;
+
+  const groupWebsites =
+    applyGrouping &&
+    dataSourceViews.filter((dsv) => isWebsite(dsv.dataSource)).length >=
+      MIN_DATA_SOURCES_PER_KIND_TO_GROUP;
+
+  return (
+    <Tree isLoading={false}>
+      {groupManaged && (
+        <Tree.Item
+          key="connected"
+          label="Connected Data"
+          visual={CloudArrowLeftRightIcon}
+          type="node"
+        >
+          {orderDatasourceViewByImportance(dataSourceViews)
+            .filter((dsv) => isManaged(dsv.dataSource))
+            .map((dataSourceView) => (
+              <DataSourceViewSelector
+                key={dataSourceView.sId}
+                owner={owner}
+                selectionConfiguration={
+                  selectionConfigurations[dataSourceView.sId] ??
+                  defaultSelectionConfiguration(dataSourceView)
+                }
+                setSelectionConfigurations={setSelectionConfigurations}
+              />
+            ))}
+        </Tree.Item>
+      )}
+
+      {orderDatasourceViewByImportance(
+        dataSourceViews.filter(
+          (dsv) =>
+            (!groupFolders && isFolder(dsv.dataSource)) ||
+            (!groupWebsites && isWebsite(dsv.dataSource)) ||
+            (!groupManaged && isManaged(dsv.dataSource))
+        )
+      ).map((dataSourceView) => (
+        <DataSourceViewSelector
+          key={dataSourceView.sId}
+          owner={owner}
+          selectionConfiguration={
+            selectionConfigurations[dataSourceView.sId] ??
+            defaultSelectionConfiguration(dataSourceView)
+          }
+          setSelectionConfigurations={setSelectionConfigurations}
+        />
+      ))}
+
+      {groupFolders && (
+        <Tree.Item key="folders" label="Files" visual={FolderIcon} type="node">
+          {dataSourceViews
+            .filter((dsv) => isFolder(dsv.dataSource))
+            .map((dataSourceView) => (
+              <DataSourceViewSelector
+                key={dataSourceView.sId}
+                owner={owner}
+                selectionConfiguration={
+                  selectionConfigurations[dataSourceView.sId] ??
+                  defaultSelectionConfiguration(dataSourceView)
+                }
+                setSelectionConfigurations={setSelectionConfigurations}
+              />
```

Comment:
Can we create a generic function for this code so we avoid duplicate?

Reply:
Are you talking about this ?

`selectionConfigurations[dataSourceView.sId] ?? defaultSelectionConfiguration(dataSourceView)`

If so, I think a generic function for a one liner is a bit over-engineering ;-)

Reply:
I meant for the whole `DataSourceViewSelector` element that very much seems the same between all the occurrences.

Reply:
If I'm understanding you correctly, considering that DataSourceViewSelector is basically already a function, I don't think adding another layer of indirection to save 3 args from repeating themselves is worth it.

Reply:
`DataSourceViewSelector` is a component and the suggestion here to have a local function to render this component ... but it's hope to you 😉.